src/libs/fileDownload/index.ts

```diff
@@ -24,7 +24,12 @@ const fileDownload: FileDownload = (url, fileName, successMessage = '', shouldOp
         return Promise.resolve();
     }
 
-    return fetch(url)
+    const fetchOptions: RequestInit = {
+        method: requestType,
+        body: formData,
+    };
+
+    return fetch(url, fetchOptions)
```

Comment:
Currently if some error occurs we are opening the the url in a new tab (line 59), lets show the generic error modal for the CSV download case

![image](https://github.com/user-attachments/assets/c6be7142-e9c7-4fbe-a1ae-5e530ad39558)